core/src/View/Kokkos_ViewLegacy.hpp

```diff
@@ -976,27 +976,27 @@ class View : public ViewTraits<DataType, Properties...> {
       std::enable_if_t<!Impl::ViewCtorProp<P...>::has_pointer,
                        typename traits::array_layout> const& arg_layout)
       : m_track(), m_map() {
+    if (is_finalized()) {
+      abort(
+          "Kokkos ERROR: View is being constructed after finalize() has been "
+          "called");
+    }
+    if (!is_initialized()) {
+      abort(
+          "Kokkos ERROR: View is being constructed before initialize() has "
+          "been called");
+    }
```

Comment:
Could we get the label out of arg_prop if it has one? Text: "Kokkos Error: View 'label' is being ..."